x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/util/StringUtils.java

```diff
@@ -354,11 +354,15 @@ public static Number parseIntegral(String string) throws InvalidArgumentExceptio
             }
             return bi;
         }
-        // try to downsize to int if possible (since that's the most common type)
-        if (bi.intValue() == bi.longValue()) { // ternary operator would always promote to Long
-            return bi.intValueExact();
-        } else {
-            return bi.longValueExact();
+        try {
+            // try to downsize to int if possible (since that's the most common type)
+            if (bi.intValue() == bi.longValue()) { // ternary operator would always promote to Long
+                return bi.intValueExact();
+            } else {
+                return bi.longValueExact();
+            }
+        } catch (ArithmeticException ae) {
```

Comment:
This works, but I'd probably check `bi` against `Long.MIN_VALUE` and throw a dedicated exception if strictly smaller. Since by the time you get here, you need to know that magnitude of positive number's been checked anyways to throw that message.